### PR TITLE
#30 course menagement create connection course has module

### PR DESCRIPTION
**Pull request by BojanMirkovic**
# CreateCourseHasModuleConnection

## Description
I have created CreateCourseHasModuleConnection endpoint and related Command and Handler with Repository method and releted tests.
I have noticed that often we are using same kind of OperationResult class in all Commands (CreateCourseResult,DeleteCourseResult...) all of them are sharing same code , so I have decide to make generic OperationResult class in Domain-Layer and to implement in Command CreateCourseHasModuleConnection .
This generic class can be easily implemented in rest of commands that way we are going to be sure that we are following DRY principal of clean code.

## Changes Made
CreateCourseHasModuleConnection endpoint, command + handler, repository, tests
DeleteModuleByModuleIdAsync is updated in oreder to delete all relations if module with that id has ben deleted

## Related Ticket or Issue
#30 
#160 
#136 

## Checklist

- [x] Code adheres to the coding conventions
- [x] Unit tests are added for new code
- [x] Existing unit tests are updated if needed
- [x] Code follows the principles of Clean Architecture
- [x] Code formatting complies with the `.editorconfig` file
- [x] Pull request template is filled out

### CODE REVIEW

File: API-Layer/Controllers/CourseController.cs [hunk 1]
```diff
@@ -55,7 +56,27 @@ public async Task<IActionResult> CreateCourse([FromBody] CreateCourseDTO courseD
                 return BadRequest(ex.Message);
             }
         }
+        [HttpPost("CreateCourseHasModulesConnection")]
```
Comment (HubTicaBB) [inline API-Layer/Controllers/CourseController.cs:59]: The route name exposes too much details about the DB infrastructure and implementation details of how courses and modules are connected. User shouldn't know about the join table and might not understand what the route does. We should have intuitive names that describe what the action means from the user's perspective, ie. adding/connect a module to a course.

I'd suggest naming the route like `course/{courseId}/add-module/{moduleId}`or `course/{courseId}/module/{moduleId}`.

**Comment by HubTicaBB** [comment]
LGTM 🙂

File: API-Layer/Controllers/CourseController.cs (hunk 1 above)
Comment (BojanMirkovic) [reply API-Layer/Controllers/CourseController.cs]: route name is changed as sugested

**Review by ElliotDahlin — APPROVED** [review]
looks good

**Pull request merged**